### DISCUSSION TITLE
Why is CoreCLR Int32.TryParse not done in the most optimal way?

### DISCUSSION THREAD

**Discussion by iamkisly** [dotnet/runtime#92591] [Q&A] [answered]
Hi all. I would like to know why [Int32.TryParse](https://github.com/dotnet/runtime/blob/af8172dc268dcaae71bfda421e88c98386c527c9/src/libraries/System.Private.CoreLib/src/System/Int32.cs#L167) is designed this way.
Why is validation carried out in this place before checking the value for null? It is logical to assume that if an null object is passed, then style validation is not needed. 
Maybe this reserves the possibility of expansion, but it looks very strange. Or, more likely, I don't understand what's going on here correctly :)

**Comment by stephentoub** [comment, marked as answer]
Because by spec an erroneous number style needs to result in an exception regardless of the string to be parsed. Otherwise, you only know that you're holding it wrong for some input strings.

**Comment by iamkisly** [comment]
OK. The answer suits me, although I think this behavior is strange :)
